filecoin-proofs/tests/regression.rs

```diff
@@ -0,0 +1,265 @@
+#[cfg(feature = "persist-regression-proofs")]
+use filecoin_proofs::{
+    MerkleTreeTrait, SealCommitOutput, SealPreCommitOutput, PUBLISHED_SECTOR_SIZES,
+};
+use storage_proofs_core::sector::SectorId;
+
+const V16_SEAL_REGRESSION_RECORDS: &str = "seal_regression_records-v16.json";
+const V18_SEAL_REGRESSION_RECORDS: &str = "seal_regression_records-v18.json";
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub(crate) struct SealRegressionRecord {
+    pub porep_config: PoRepConfig,
+    pub comm_r: Commitment,
+    pub comm_d: Commitment,
+    pub prover_id: ProverId,
+    pub sector_id: SectorId,
+    pub ticket: [u8; 32],
+    pub seed: [u8; 32],
+    pub proof: Vec<u8>,
+}
+
+#[cfg(feature = "persist-regression-proofs")]
+#[allow(dead_code)]
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn persist_generated_proof_for_regression_testing<Tree: 'static + MerkleTreeTrait>(
+    config: &PoRepConfig,
+    prover_id: ProverId,
+    sector_id: SectorId,
+    ticket: [u8; 32],
+    seed: [u8; 32],
+    pre_commit_output: &SealPreCommitOutput,
+    commit_output: &SealCommitOutput,
+) -> Result<()> {
+    use std::io::Write;
+
+    use anyhow::ensure;
+    use file_lock::{FileLock, FileOptions};
+
+    const DEFAULT_SEAL_REGRESSION_RECORD: &str = "seal_regression_records.json";
+
+    // First, make sure that we're only persisting sector sizes with
+    // published parameters so that everyone can verify them properly
+    let sector_size: u64 = config.sector_size.into();
+
+    if PUBLISHED_SECTOR_SIZES.contains(&sector_size) {
+        // When generating a new record file, use a default path for persisting them
+        let output_path = Path::new(&std::env::var("CARGO_MANIFEST_DIR")?)
+            .join("tests")
+            .join(DEFAULT_SEAL_REGRESSION_RECORD);
+
+        let comm_d = pre_commit_output.comm_d;
+        let comm_r = pre_commit_output.comm_r;
+
+        // Then verify this proof before persisting it
+        let verified = verify_seal::<Tree>(
+            config,
+            comm_r,
+            comm_d,
+            prover_id,
+            sector_id,
+            ticket,
+            seed,
+            &commit_output.proof,
+        )?;
+        ensure!(verified, "failed to verify valid seal");
+
+        let record = SealRegressionRecord {
+            porep_config: config.clone(),
+            comm_r,
+            comm_d,
+            prover_id,
+            sector_id,
+            ticket,
+            seed,
+            proof: commit_output.proof.clone(),
+        };
+
+        let json = serde_json::to_string(&record)?;
+
+        // When appending to an existing file, the file lock is required
+        // to avoid corruption when multiple tests running in parallel are
+        // persisting generated proofs
+        let options = FileOptions::new().write(true).create(true).append(true);
+        let mut filelock = FileLock::lock(output_path, true /* block */, options)?;
```

Comment:
Just flagging this line in case the "block" comment was meant to be removed before merging.

Reply:
That was intentional to give context to what the hard coded `true` value is specifying